client/src/main/java/io/pravega/client/stream/impl/LargeEventWriter.java

```diff
@@ -179,65 +176,22 @@ private void write(Segment parentSegment, List<ByteBuf> payloads, RawClient clie
                     Unpooled.wrappedBuffer(payload),
                     requestId);
             expectedOffset += payload.readableBytes();
-            val reply = client.sendRequest(requestId, request);
-            failFast(futures, created.getSegment());
-            futures.add(reply);
+            // Wait for Appends to complete. Writing in parallel Appends that are expected to be sequential does not help.
```

Comment:
This isn't at all true. We support append pipelining on both the client and the server. This will avoid round trips for each of them and allow them to be written sequentially with the client starting the next write as soon as the previous has finished. 

Please restore the failfast/join at the end.